#!/usr/bin/env python
PACKAGE = "icp"

# When copying to make your own:
#  - Different parameters
#  - Add cfg file to CMakeLists file, in generate_dynamic_reconfigure_options

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Name, Type, Level, Description, Default, Min, Max

# Actual Parameters
gen.add("test_angle_increment", double_t, 0, "Angle Increment", 0.001, 0.0000001, 0.1)
gen.add("test_tx", double_t, 0, "Test Trans in x", 0, -5, 5)
gen.add("test_ty", double_t, 0, "Test Trans in x", 0, -5, 5)
gen.add("test_theta", double_t, 0, "Test Trans in theta", 0, -0.5, 0.5)
gen.add("outer_loop_count", int_t, 0, "Outer ICP Iter #", 3, 1, 100)

exit(gen.generate(PACKAGE, "icp", "ICP"))